### PR TITLE
[#55771] Warn user that the type is not available when moving a work package

### PR DESCRIPTION
**Pull request by aaron-contreras**
Displays a warning toast letting the user know to activate the current types of the work packages under move in the target project and making the "auto-reassign" behavior known to them so they can better take a decision when performing moves.

See: https://community.openproject.org/work_packages/55771

### CODE REVIEW

**Review by dombesz — COMMENTED** [review]
The warning seems to be coming on, even if I don't change anything on the move form yet.
<details><summary>Bug Demo</summary>
<p>

Jun-20-2024 10-16-32


</p>
</details> 


<details><summary>Just an observation,the toast seems to have no space on top. Please ignore it if that's the intended look.</summary>
<p>

image


</p>
</details>

**Comment by aaron-contreras** [comment]
> The warning seems to be coming on, even if I don't change anything on the move form yet.

I see this as being correct though, it's warning you if you haven't done anything because not doing anything will cause the type to be automatically re-assigned.

> Just an observation,the toast seems to have no space on top. Please ignore it if that's the intended look.

There were no visuals for it but I wanted to make it seem tied to the field above and not just floating around 😅 . Do you have another idea?

File: app/controllers/work_packages/moves_controller.rb [hunk 1]
```diff
@@ -114,8 +115,9 @@ def prepare_for_work_package_move
     @allowed_projects = WorkPackage.allowed_target_projects_on_move(current_user)
     @target_project = @allowed_projects.detect { |p| p.id.to_s == params[:new_project_id].to_s } if params[:new_project_id]
     @target_project ||= @project
-    @types = @target_project.types
+    @types = @target_project.types.order(:position)
     @target_type = @types.find { |t| t.id.to_s == params[:new_type_id].to_s }
+    @unavailable_type_in_target_project = @types.exclude?(@target_type)
```
Comment (dombesz) [inline app/controllers/work_packages/moves_controller.rb]: If the `@target_project == @project` then the `@unavailable_type_in_target_project` should be `false`.

Furthermore, when I move a WP with type task, that is enabled in the target project too while leaving the `Type` on `(No change)`, the `params[:new_type_id]` will not be set, which in turn makes the warning to be shown. I believe that is incorrect.

Comment (aaron-contreras) [reply app/controllers/work_packages/moves_controller.rb]: Nice catch. I understand now what you mean

Addressed in latest commit!

**Review by dombesz — APPROVED** [review]
Nice job Aaron! 👍

**Pull request merged**